app/src/main/java/ch/epfl/sdp/appart/SimpleUserProfileActivity.java

```diff
@@ -80,10 +94,59 @@ protected void onCreate(Bundle savedInstanceState) {
     }
 
     /**
-     * closes activity when back button pressed on UI
+     * Contact announcer.
      */
-    public void contactAdUser(View view) {
-        // TODO: send message to user
+    public void openEmailOrPhone(View view){
+        if(!advertiserUser.getPhoneNumber().isEmpty()) {
+            AlertDialog.Builder builder = new AlertDialog.Builder(this);
+            builder.setTitle("How did you prefer contact the announcer ?");
+            builder.setPositiveButton("Contact via Email", (dialog, which) -> onEmail());
+            builder.setNeutralButton("Contact via phone number", (dialog, which) -> onCall());
+            AlertDialog dialog = builder.create();
+            dialog.show();
+        } else {
+            onEmail();
+        }
+    }
+
+    private void onEmail() {
+        Intent intent = new Intent(Intent.ACTION_SENDTO);
+        intent.setData(Uri.parse("mailto:"));
+        intent.putExtra(Intent.EXTRA_EMAIL,  new String[]{advertiserUser.getUserEmail()});
+        intent.putExtra(Intent.EXTRA_SUBJECT, "Rent apartment");
+        if (intent.resolveActivity(getPackageManager()) != null) {
+            startActivity(intent);
+        } else {
+            makeText(this, "Error open email, try again",Toast.LENGTH_SHORT).show();
+
+        }
+    }
+    private void onCall() {
+        int permissionCheck = ContextCompat.checkSelfPermission(this, Manifest.permission.CALL_PHONE);
+        if (permissionCheck != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(
+                this,
+                new String[]{Manifest.permission.CALL_PHONE},
+                123);
+        } else {
```

Comment:
Maybe in a refactor PR we could modify this code to follow the same way it is structured in other cases where we ask for permissions

Reply:
Good catch @rovati ! This should be implemented in the `PermissionRequest`. Other thing to add to the final refactor.